app/src/main/java/com/mathewsachin/fategrandautomata/imaging/DroidCvPattern.kt

```diff
@@ -27,24 +27,28 @@ class DroidCvPattern(
         fun makeMat(stream: InputStream): MatWithAlpha {
             val byteArray = stream.readBytes()
             MatOfByte(*byteArray).use {
-                val grayScale = Imgcodecs.imdecode(it, Imgcodecs.IMREAD_GRAYSCALE)
-                var alphaChannel: Mat? = null
-
-                Imgcodecs.imdecode(it, Imgcodecs.IMREAD_UNCHANGED).use { original ->
-                    // RGBA, extract alpha
-                    if (original.channels() == 4) {
-                        alphaChannel =
-                            Mat().apply { Core.extractChannel(original, this, 3) }
-                        val minMax = Core.minMaxLoc(alphaChannel)
-                        if (minMax.minVal.equals(255.0)) {
-                            //every pixel has 0 transparency, alpha is useless
-                            alphaChannel?.release()
-                            alphaChannel = null
+                Imgcodecs.imdecode(it, Imgcodecs.IMREAD_UNCHANGED).use { decoded ->
+                    val grayScale = Mat()
+                    var alphaChannel: Mat? = null
+
+                    when (decoded.channels()) {
+                        4 -> {
+                            // RGBA, extract alpha
+                            alphaChannel =
+                                Mat().apply { Core.extractChannel(decoded, this, 3) }
+                            val minMax = Core.minMaxLoc(alphaChannel)
+                            if (minMax.minVal.equals(255.0)) {
+                                //every pixel has 0 transparency, alpha is useless
+                                alphaChannel.release()
+                                alphaChannel = null
+                            }
+                            Imgproc.cvtColor(decoded, grayScale, Imgproc.COLOR_BGRA2GRAY)
                         }
+                        3 -> Imgproc.cvtColor(decoded, grayScale, Imgproc.COLOR_BGRA2GRAY)
```

Comment:
RGBA would be for 4 channels. RGB for 3.

Reply:
Don't you mean BGR? According to the internet, OpenCV reads images in BGR format: https://stackoverflow.com/a/59851524/5971497

That's not the only place where I've found this info.